ompi/mca/common/ompio/common_ompio_buffer.c

```diff
@@ -47,7 +48,11 @@ void mca_common_ompio_check_gpu_buf ( ompio_file_t *fh, const void *buf, int *is
 
     *is_gpu=0;
     *is_managed=0;
-    
+
+    if (fh->f_fh->f_flags & OMPI_FILE_ASSERT_NO_ACCEL_BUF) {
```

Comment:
Could we do the same for `OMPI_INFO_MEMKIND_ASSERT_ACCEL_DEVICE_ONLY` and skip the check of the addr below?

Reply:
we might be able to do that potentially. The downside that I see however in the `DEVICE_ONLY` mode is that we loose the ability to retrieve the device_id where the buffer is located. Not sure whether that's relevant or not, but that was one of the reasons why I decided not to pursue that in the first stage.